File: 双指针/LC 283移动零.cpp
//LC 283移动零 双指针做法
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i0 = 0;
        for (int& x : nums) { // 注意 x 是引用
            if (x) {
                swap(x, nums[i0]);
                i0++;
            }
        }
    }
};

File: 双指针/LC 283移动零.java
class Solution {
    public void moveZeroes(int[] nums) {
        int index0=0;
        for(int index=0 ; index<nums.length;index++){
            if(nums[index]!=0){
                int temp = nums[index];
                nums[index] = nums[index0];
                nums[index0] = temp;
                index0++;
            }
        }
    }
}